pumps/resurface_test.go

```diff
@@ -0,0 +1,333 @@
+	rawReq string = "R0VUIGdldCBIVFRQLzEuMQ0KSG9zdDogbG9jYWxob3N0OjgwODANClVzZXItQWdlbnQ6IE1vemls" +
+		"bGEvNS4wIChYMTE7IFVidW50dTsgTGludXggeDg2XzY0OyBydjo5MS4wKSBHZWNrby8yMDEwMDEwMSBGaXJlZm94" +
+		"LzkxLjANCkFjY2VwdDogdGV4dC9odG1sLGFwcGxpY2F0aW9uL3hodG1sK3htbCxhcHBsaWNhdGlvbi94bWw7cT0w" +
+		"LjksaW1hZ2Uvd2VicCwqLyo7cT0wLjgNCkFjY2VwdC1FbmNvZGluZzogZ3ppcCwgZGVmbGF0ZQ0KQWNjZXB0LUxh" +
+		"bmd1YWdlOiBlbi1VUyxlbjtxPTAuNQ0KU2VjLUZldGNoLURlc3Q6IGRvY3VtZW50DQpTZWMtRmV0Y2gtTW9kZTog" +
+		"bmF2aWdhdGUNClNlYy1GZXRjaC1TaXRlOiBub25lDQo="
+
+	rawResp string = "SFRUUC8xLjEgMjAwIE9LDQpDb250ZW50LUxlbmd0aDogNDI5DQpBY2Nlc3MtQ29udHJvbC1BbGxv" +
+		"dy1DcmVkZW50aWFsczogdHJ1ZQ0KQWNjZXNzLUNvbnRyb2wtQWxsb3ctT3JpZ2luOiAqDQpDb250ZW50LVR5cGU6I" +
+		"GFwcGxpY2F0aW9uL2pzb24NCkRhdGU6IFR1ZSwgMTYgTWF5IDIwMjAgMjA6NDA6NDUgR01UDQpTZXJ2ZXI6IGd1bm" +
+		"ljb3JuLzE5LjkuMA0KWC1SYXRlbGltaXQtTGltaXQ6IDANClgtUmF0ZWxpbWl0LVJlbWFpbmluZzogMA0KWC1SYXR" +
+		"lbGltaXQtUmVzZXQ6IDANCg0Kew0KICAic2xpZGVzaG93Ijogew0KICAgICJhdXRob3IiOiAiWW91cnMgVHJ1bHki" +
+		"LCANCiAgICAiZGF0ZSI6ICJkYXRlIG9mIHB1YmxpY2F0aW9uIiwgDQogICAgInNsaWRlcyI6IFsNCiAgICAgIHsNC" +
+		"iAgICAgICAgInRpdGxlIjogIldha2UgdXAgdG8gV29uZGVyV2lkZ2V0cyEiLCANCiAgICAgICAgInR5cGUiOiAiYW" +
+		"xsIg0KICAgICAgfSwgDQogICAgICB7DQogICAgICAgICJpdGVtcyI6IFsNCiAgICAgICAgICAiV2h5IDxlbT5Xb25" +
+		"kZXJXaWRnZXRzPC9lbT4gYXJlIGdyZWF0IiwgDQogICAgICAgICAgIldobyA8ZW0+YnV5czwvZW0+IFdvbmRlcldp" +
+		"ZGdldHMiDQogICAgICAgIF0sIA0KICAgICAgICAidGl0bGUiOiAiT3ZlcnZpZXciLCANCiAgICAgICAgInR5cGUiO" +
+		"iAiYWxsIg0KICAgICAgfQ0KICAgIF0sIA0KICAgICJ0aXRsZSI6ICJTYW1wbGUgU2xpZGUgU2hvdyINCiAgfQ0KfQ" +
+		"=="
+
+	rawRespOneChunk string = "SFRUUC8xLjEgMjAwIE9LDQpDb250ZW50LVR5cGU6IHRleHQvcGxhaW4NClRyYW5zZmVy" +
+		"LUVuY29kaW5nOiBjaHVua2VkDQoNCjcNCg0KTW96aWxsYQ0KDQoxMQ0KDQpEZXZlbG9wZXIgTmV0d29yaw0KDQowD" +
+		"QoNCg0KDQo="
+
+	rawRespChunks string = "SFRUUC8xLjEgMjAwIE9LDQpUcmFuc2Zlci1FbmNvZGluZzogY2h1bmtlZA0KQ29udGVudC" +
+		"1UeXBlOiB0ZXh0L2h0bWwNCg0KYw0KPGgxPmdvITwvaDE+DQoNCjFiDQo8aDE+Zmlyc3QgY2h1bmsgbG9hZGVkPC9" +
+		"oMT4NCg0KMmENCjxoMT5zZWNvbmQgY2h1bmsgbG9hZGVkIGFuZCBkaXNwbGF5ZWQ8L2gxPg0KDQoyOQ0KPGgxPnRo" +
+		"aXJkIGNodW5rIGxvYWRlZCBhbmQgZGlzcGxheWVkPC9oMT4NCg0KMA0K"
+
+	rawRespChunksTrailer string = "SFRUUC8xLjEgMjAwIE9LDQpUcmFuc2Zlci1FbmNvZGluZzogY2h1bmtlZA0KQ29" +
+		"udGVudC1UeXBlOiB0ZXh0L2h0bWwNClRyYWlsZXI6IEV4cGlyZXMNCg0KYw0KPGgxPmdvITwvaDE+DQoNCjFiDQo8" +
+		"aDE+Zmlyc3QgY2h1bmsgbG9hZGVkPC9oMT4NCg0KMmENCjxoMT5zZWNvbmQgY2h1bmsgbG9hZGVkIGFuZCBkaXNwb" +
+		"GF5ZWQ8L2gxPg0KDQoyOQ0KPGgxPnRoaXJkIGNodW5rIGxvYWRlZCBhbmQgZGlzcGxheWVkPC9oMT4NCg0KMA0KRX" +
+		"hwaXJlczogV2VkLCAyMSBPY3QgMjAxNSAwNzoyODowMCBHTVQNCg0K"
+)
+
+func SetUp(t *testing.T, url string, queue []string, rules string) (*ResurfacePump, map[string]interface{}) {
+	pmp := ResurfacePump{}
+	cfg := make(map[string]interface{})
+	cfg["capture_url"] = url
+	cfg["queue"] = queue
+	cfg["rules"] = rules
+
+	err := pmp.Init(cfg)
+	if err != nil {
+		t.Fatal("Problem initializing "+pmp.GetName()+": ", err)
```

Comment:
we can use assert.NotNil here

Reply:
Thank you, I just modified it accordingly.